src/shrinkray/tui.py

```diff
@@ -424,6 +637,191 @@ def compose(self) -> ComposeResult:
             yield Static("[dim]Press any key to close[/dim]")
 
 
+class ExpandedBoxModal(ModalScreen[None]):
+    """Modal screen showing an expanded view of a content box."""
+
+    CSS = """
+    ExpandedBoxModal {
+        align: center middle;
+    }
+
+    ExpandedBoxModal > Vertical {
+        width: 95%;
+        height: 90%;
+        background: $panel;
+        border: thick $primary;
+        padding: 0 1 1 1;
+    }
+
+    ExpandedBoxModal #expanded-title {
+        text-align: center;
+        text-style: bold;
+        height: auto;
+        width: 100%;
+        border-bottom: solid $primary;
+        padding: 0;
+        margin-bottom: 1;
+    }
+
+    ExpandedBoxModal VerticalScroll {
+        width: 100%;
+        height: 1fr;
+    }
+
+    ExpandedBoxModal #expanded-content {
+        width: 100%;
+    }
+
+    ExpandedBoxModal #expanded-graph {
+        width: 100%;
+        height: 1fr;
+    }
+    """
+
+    BINDINGS = [
+        ("escape,enter,q", "dismiss", "Close"),
+    ]
+
+    def __init__(
+        self, title: str, content_widget_id: str, file_path: str | None = None
+    ) -> None:
+        super().__init__()
+        self._title = title
+        self._content_widget_id = content_widget_id
+        self._file_path = file_path
+
+    def _read_file_with_retry(self, file_path: str, max_retries: int = 3) -> str:
+        """Read file content with retry for transient errors during reduction.
+
+        The file may be briefly unavailable while the reducer is writing a new
+        version. Retry a few times with a short delay to handle this.
+        """
+        last_error: OSError | None = None
+        for attempt in range(max_retries):
+            try:
+                with open(file_path, "rb") as f:
+                    raw_content = f.read()
+                # Try to decode as text, fall back to hex if binary
+                try:
+                    return raw_content.decode("utf-8")
+                except UnicodeDecodeError:
+                    return "[Binary content - hex display]\n\n" + raw_content.hex()
+            except OSError as e:
+                last_error = e
+                if attempt < max_retries - 1:
+                    time.sleep(0.05)  # 50ms delay between retries
```

Comment:
This surely should not be a synchronous sleep given that everything else is async.

Please add a lint rule that asserts that nothing in `src` ever calls `time.sleep`